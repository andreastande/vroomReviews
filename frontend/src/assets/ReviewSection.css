.popup {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.2);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
}

.popup-inner {
  display: grid;
  grid-template-columns: 1fr;
  position: relative;
  border-radius: 0.5rem;
  padding: 2rem;
  height: fit-content;
  min-width: 50%;
  width: fit-content;
  background-color: #252525;
}

.delete-popup {
  display: flex;
  gap: 2rem;
}

.review-buttons {
  display: flex;
  gap: 2rem;
  margin-top: 1rem;
}

.text-area {
  background-color: #202020;
  padding: 0.75rem;
  margin-top: 1rem;
  border-radius: 0.5rem;
  border: 1px solid silver;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.text {
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: silver;
}

.review-buttons {
  display: flex;
}

.reviews {
  margin-top: 3rem;
}

.current-user-review {
  display: flex;
  gap: 2rem;
  margin: 2rem 0;
  border-top: 0.2rem solid grey;
  padding-top: 1rem;
}

.user-review {
  margin-top: 1rem;
  border-top: 0.2rem solid grey;
  padding-top: 1rem;
}

.delete-review {
  font-size: medium;
  color: #bd1b12;
  cursor: pointer;
  margin-top: 1rem;
}

.reviewer {
  font-size: large;
  color: rgb(109, 109, 109);
  margin-left: 1rem;
}

@media (orientation: portrait) {
  .review-buttons {
    flex-direction: column;
  }
}
