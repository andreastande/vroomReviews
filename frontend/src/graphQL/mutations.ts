import { gql } from '@apollo/client';

export const ADD_FAVORITE_CAR = gql`
  mutation AddFavoriteCar($userID: Int!, $car: String!) {
    addFavorite(userID: $userID, car: $car) {
      userID
      car {
        company
        model
      }
    }
  }
`;

export const REMOVE_FAVORITE_CAR = gql`
  mutation RemoveFavoriteCar($userID: Int!, $car: String!) {
    removeFavorite(userID: $userID, car: $car) {
      userID
      car {
        company
        model
      }
    }
  }
`;

export const ADD_REVIEW = gql`
  mutation AddReview($userID: Int!, $car: String!, $rating: Int!, $review: String!, $username: String!) {
    addReview(userID: $userID, car: $car, rating: $rating, review: $review, username: $username) {
      rating
      review
      username
    }
  }
`;

export const REMOVE_REVIEW = gql`
  mutation RemoveReview($userID: Int!, $car: String!) {
    removeReview(userID: $userID, car: $car) {
      rating
      review
      username
    }
  }
`;

export const ADD_USER = gql`
  mutation AddUser($userID: Int!) {
    addUser(userID: $userID) {
      userID
    }
  }
`;
