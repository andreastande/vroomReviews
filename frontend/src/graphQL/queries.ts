import { gql } from '@apollo/client';

export const GET_CAR = gql`
  query GetCar($company: String!, $model: String!) {
    car(company: $company, model: $model) {
      id
      company
      model
      image
      horsepower
      transmissionType
      drivetrain
      numOfDoors
      price
      year
      carBody
      engineType
      numOfCylinders
      rating
    }
  }
`;

export const GET_CARS_BY_COMPANY = gql`
  query GetCarsByCompany($company: String!) {
    carsByCompany(company: $company) {
      id
      company
      model
      image
    }
  }
`;

export const GET_CARS = gql`
  query GetCars(
    $filters: carsFilters
    $offset: Int
    $orderBy: orderByArg
    $searchTerm: String
    $limit: Int
  ) {
    cars(
      filters: $filters
      offset: $offset
      orderBy: $orderBy
      searchTerm: $searchTerm
      limit: $limit
    ) {
      company
      model
      image
    }
  }
`;

export const GET_FAVORITE_CARS = gql`
  query GetFavoriteCars($userID: Int!) {
    favoriteCars(userID: $userID) {
      car {
        id
        company
        model
        image
      }
    }
  }
`;

export const GET_CAR_REVIEWS = gql`
  query GetCarReviews($car: ID!) {
    carReviews(car: $car) {
      userID
      rating
      review
      username
    }
  }
`;

export const GET_USER_REVIEWS = gql`
  query GetUserReviews($userID: Int!) {
    userReviews(userID: $userID) {
      rating
      review
      car {
        company
        model
        image
      }
    }
  }
`;

export const GET_USER_REVIEW_FOR_CAR = gql`
  query GetUserReviewForCar($userID: Int!, $car: ID!) {
    userReviewForCar(userID: $userID, car: $car) {
      userID
      rating
      review
      username
    }
  }
`;

export const GET_USER_COUNT = gql`
  query GetUserCount {
    users
  }
`;

export const GET_COMPANIES = gql`
  query GetCompanies {
    companies {
      name
      logo
    }
  }
`;