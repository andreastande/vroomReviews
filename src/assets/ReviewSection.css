.popup {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.2);

  display: flex;
  justify-content: center;
  align-items: center;
}

.popup-inner {
  display: grid;
  grid-template-columns: 1fr;
  position: relative;
  border-radius: 8px;
  padding: 32px;
  height: 300px;
  width: 640px;
  background-color: #f6f9ff;
}

.text-area {
  padding: 0.75rem;
  margin-top: 1rem;
  border-radius: 0.5rem;
  background-color: #ffffff;
  box-shadow:
    0 1px 3px 0 rgb(0 0 0 / 0.1),
    0 1px 2px -1px rgb(0 0 0 / 0.1);
  align-items: center;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.review-buttons {
  display: flex;
}

.submit {
  margin-right: 2rem;
}

.delete {
  background-color: #ff0000;
}

.user-review {
  border: 0;
  margin-top: 20px;
  border-top: 2px solid grey;
  margin-right: 10px;
  padding-top: 10px;
}

.edit-underline {
  font-size: medium;
  color: #dd7973;
  cursor: pointer;
  margin-left: 1rem;
}

.review-section {
  margin-top: 40px;
  margin-right: 40px;
  padding-top: 10px;
  border-top: 2px solid grey;
}

.review-section text {
  width: 75%;
  float: right;
}

.reviewer {
  font-size: large;
  color: rgb(109, 109, 109);
  margin-left: 1rem;
}
