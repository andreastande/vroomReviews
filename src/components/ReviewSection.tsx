import React, { useState, MouseEventHandler } from 'react';
import { StarRating } from 'star-rating-react-ts';
import '../assets/ReviewSection.css';

type Review = {
  id: string;
  rating: number;
  comment: string;
  userID: string;
  userName: string;
  carID: number;
};

const ReviewSection = ({
  userReview,
  reviews,
}: {
  userReview: Review | undefined;
  reviews: Review[];
}) => {
  const amountOfRatingsForBook = reviews.length;

  const [reviewCarPopup, setReviewCarPopup] = useState(false);
  const [visibleDeletePopup, setVisibleDeletePopup] = useState(false);
  const [reviewAdded, setReviewAdded] = useState(false);
  const [rating, setRating] = useState(0);
  const [reviewText, setReviewText] = useState('');

  const closeOrOpen: MouseEventHandler<HTMLDivElement> = (e) => {
    const isClose = (e.target as HTMLElement).closest('#popup');
    if (!isClose) {
      setReviewCarPopup(false);
    }
  };

  const handleReviewTextChange = (
    event: React.ChangeEvent<HTMLTextAreaElement>,
  ) => {
    setReviewText(event.target.value);
    event.target.style.height = 'auto';
    event.target.style.height = event.target.scrollHeight + 'px';
  };

  function handleReviewSubmit() {
    // Add review to database
    setReviewAdded(true);
  }

  function handleEdit() {
    if (userReview) {
      setReviewText(userReview.comment);
      setRating(userReview.rating);
    }
    setReviewCarPopup(true);
  }

  function handleDeleteConfirm(review: Review) {
    console.log(review);
    // Delete review from database
    setVisibleDeletePopup(false);
    setReviewCarPopup(false);
    setReviewAdded(false);
    setReviewText('');
    setRating(0);
  }

  return (
    <div>
      <div>
        {!userReview && !reviewAdded ? (
          <button
            className="reviewButton"
            onClick={() => setReviewCarPopup(true)}
          >
            Review this car
          </button>
        ) : null}
      </div>
      {reviewCarPopup ? (
        <div>
          <div className="popup" onClick={closeOrOpen}>
            <div className="popup-inner" id="popup">
              <StarRating
                initialRating={rating}
                onClick={(rating) => {
                  setRating(rating);
                }}
              />
              <textarea
                className="text-area"
                value={reviewText}
                onChange={handleReviewTextChange}
                placeholder="Add a review to your rating"
                cols={28}
                style={{ height: 'auto', minHeight: '100px' }}
              />
              <div className="review-buttons">
                <button onClick={handleReviewSubmit} className="submit">
                  Submit review
                </button>
                {userReview ? (
                  <button
                    onClick={() => setVisibleDeletePopup(true)}
                    className="delete"
                  >
                    Delete review
                  </button>
                ) : null}
              </div>
            </div>
          </div>
        </div>
      ) : null}
      {visibleDeletePopup ? (
        <div>
          <div className="popup" onClick={closeOrOpen}>
            <div className="popup-inner" id="popup">
              <p>Are you sure you want to delete this review? </p>
              <button onClick={() => handleDeleteConfirm(userReview!)}>
                Confirm
              </button>
              <button onClick={() => setVisibleDeletePopup(false)}>
                Cancel
              </button>
            </div>
          </div>
        </div>
      ) : null}
      <div>
        {amountOfRatingsForBook === 0 ? (
          <p>There are currently no reviews for this book</p>
        ) : (
          <h1>Reviews</h1>
        )}
        {userReview ? (
          <div className="current-user-review">
            <p> Your review: </p>
            <div>
              <StarRating
                theme={{ size: 30 }}
                readOnly={true}
                initialRating={userReview?.rating}
              />
              <p>{userReview?.comment}</p>
            </div>
            <u className="edit" onClick={handleEdit}>
              edit
            </u>
          </div>
        ) : null}
        {reviews.map((review) => (
          <div key={review.userID}>
            {review.userID !== userReview?.userID ? (
              <div className="user-review">
                <StarRating
                  theme={{ size: 30 }}
                  readOnly={true}
                  initialRating={review.rating}
                />
                <p>{review.comment}</p>
                <p className="reviewer">
                  Reviewed by <i>{review.userName}</i>{' '}
                </p>
              </div>
            ) : null}
          </div>
        ))}
      </div>
    </div>
  );
};

export default ReviewSection;
