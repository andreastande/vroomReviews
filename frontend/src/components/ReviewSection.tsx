import React, { useState, MouseEventHandler } from 'react';
import { StarRating } from 'star-rating-react-ts';
import '../assets/ReviewSection.css';
import { Review } from '../types/Review';
import { useMutation } from '@apollo/client';
import { ADD_REVIEW } from '../graphQL/mutations';
import { REMOVE_REVIEW } from '../graphQL/mutations';
import { CircularProgress } from '@mui/material';
import AlertPopup from './AlertPopup';
import {
  GET_CAR_REVIEWS,
  GET_USER_REVIEWS,
  GET_USER_REVIEW_FOR_CAR,
  GET_CAR
} from '../graphQL/queries';

const ReviewSection = ({
  userReview,
  reviews,
  carID,
}: {
  userReview: Review | undefined;
  reviews: Review[];
  carID: string;
}) => {
  const userID = Number(localStorage.getItem('userID'));

  const [alertMessage, setAlertMessage] = useState('');
  const [reviewCarPopup, setReviewCarPopup] = useState(false);
  const [visibleDeletePopup, setVisibleDeletePopup] = useState(false);
  const [username, setUsername] = useState(userReview?.username || '');
  const [reviewAdded, setReviewAdded] = useState(false);
  const [rating, setRating] = useState(userReview?.rating || 0);
  const [reviewText, setReviewText] = useState(userReview?.review || '');
  const [alertVisible, setAlertVisible] = useState(false);
  const amountOfReviews = reviews.length;

  // Add or remove review from database, refetch queries to update cache with reviews

  const [addReview, { loading: addLoading, error: addError }] = useMutation(
    ADD_REVIEW,
    {
      variables: {
        userID: userID,
        car: carID,
        rating: rating,
        review: reviewText,
        username: username,
      },
      refetchQueries: [
        GET_USER_REVIEW_FOR_CAR,
        GET_CAR_REVIEWS,
        GET_USER_REVIEWS,
        GET_CAR,
      ],
    },
  );

  const [removeReview, { loading: removeLoading, error: removeError }] =
    useMutation(REMOVE_REVIEW, {
      variables: {
        userID: userID,
        car: carID,
      },
      refetchQueries: [
        GET_USER_REVIEW_FOR_CAR,
        GET_CAR_REVIEWS,
        GET_USER_REVIEWS,
        GET_CAR,
      ],
    });

  // Close popup when clicking outside of it
  const closeOrOpen: MouseEventHandler<HTMLDivElement> = (e) => {
    const isClose = (e.target as HTMLElement).closest('#popup');
    if (!isClose) {
      setReviewCarPopup(false);
      setVisibleDeletePopup(false);
    }
  };

  // Set review text to text inputed and resize textarea
  const handleReviewTextChange = (
    event: React.ChangeEvent<HTMLTextAreaElement>,
  ) => {
    setReviewText(event.target.value);
    event.target.style.height = 'auto';
    event.target.style.height = event.target.scrollHeight + 'px';
  };

  // Add review to database
  function handleReviewSubmit() {
    addReview();
    setReviewAdded(true);
    setReviewCarPopup(false);
    setAlertMessage('Successfully added review!');
    setAlertVisible(true);
  }

  // Delete review from database
  function handleDeleteConfirm() {
    removeReview();
    setReviewAdded(false);
    setVisibleDeletePopup(false);
    setRating(0);
    setAlertMessage('Successfully deleted review!');
    setAlertVisible(true);
  }

  if (addLoading || removeLoading) return <CircularProgress color="warning" />;
  if (addError || removeError) setAlertMessage('Something went wrong!');

  return (
    <div>
      <div>
        {!userReview && !reviewAdded ? (
          <button className="button" onClick={() => setReviewCarPopup(true)}>
            Review this car
          </button>
        ) : null}
      </div>
      {reviewCarPopup ? (
        <div>
          <div className="popup" onClick={closeOrOpen}>
            <div className="popup-inner" id="popup">
              <StarRating
                initialRating={rating}
                onClick={(rating) => {
                  setRating(rating);
                }}
              />
              <textarea
                className="text-area"
                onChange={handleReviewTextChange}
                placeholder="Add a review to your rating"
                cols={28}
                style={{ height: 'auto', minHeight: '100px' }}
              />
              <input
                className="text-area"
                placeholder="name"
                onChange={(e) => setUsername(e.target.value)}
              />
              <div className="review-buttons">
                <button onClick={handleReviewSubmit} className="button">
                  Submit review
                </button>
              </div>
            </div>
          </div>
        </div>
      ) : null}
      {visibleDeletePopup ? (
        <div>
          <div className="popup" onClick={closeOrOpen}>
            <div className="popup-inner" id="popup">
              <p className="text">
                Are you sure you want to delete this review?{' '}
              </p>
              <button className="button" onClick={() => handleDeleteConfirm()}>
                Confirm
              </button>
              <button
                className="button"
                onClick={() => setVisibleDeletePopup(false)}
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
      ) : null}
      <div className="reviews">
        {amountOfReviews > 0 ? (
          <h1>Reviews ({amountOfReviews})</h1>
        ) : (
          <h2>There are no reviews yet for this car</h2>
        )}
        {userReview || reviewAdded ? (
          <div className="current-user-review">
            <p> Your review: </p>
            <div>
              <StarRating
                theme={{ size: 30 }}
                readOnly={true}
                initialRating={rating}
              />
              <p>{reviewText}</p>
            </div>
            <u
              className="delete-review"
              onClick={() => setVisibleDeletePopup(true)}
            >
              delete
            </u>
          </div>
        ) : null}
        {reviews.map((review) => (
          <div key={review.userID}>
            {review.userID !== userReview?.userID ? (
              <div className="user-review">
                <StarRating
                  theme={{ size: 30 }}
                  readOnly={true}
                  initialRating={review.rating}
                />
                <p>{review.review}</p>
                <p className="reviewer">Reviewed by {review.username}</p>
              </div>
            ) : null}
          </div>
        ))}
        <AlertPopup visible={alertVisible} message={alertMessage} />
      </div>
    </div>
  );
};

export default ReviewSection;
