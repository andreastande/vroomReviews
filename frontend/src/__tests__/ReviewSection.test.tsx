import { render, fireEvent, waitFor } from '@testing-library/react';
import { test, expect } from 'vitest';
import ReviewSection from '../components/ReviewSection'; // Update the path accordingly
import { MockedProvider } from '@apollo/client/testing';
import { ADD_REVIEW } from '../graphQL/mutations';
import userEvent from '@testing-library/user-event';

const mockProps = {
  userReview: {
    id: '1',
    rating: 5,
    review: 'Test review',
    username: 'Test user1',
    car: '123',
    userID: 123,
  },
  reviews: [],
  carID: '123',
};

const mocks = [
  {
    request: {
      query: ADD_REVIEW,
      variables: {
        userID: 0,
        car: '1',
        rating: 5,
        review: 'Test review',
        username: 'Test user2',
      },
    },
    result: {
      data: {
        rating: 5,
        review: 'Test review',
        username: 'Test user2',
      },
    },
  },
];

describe('ReviewSection Component', () => {
  test('renders the component without errors', () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReviewSection
          userReview={mockProps.userReview}
          reviews={[]}
          carID="123"
        />
      </MockedProvider>,
    );
    expect(container).toBeTruthy();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReviewSection
          userReview={mockProps.userReview}
          reviews={[]}
          carID="123"
        />
      </MockedProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays the review form when "Review this car" button is clicked', () => {
    const { getByText, getByPlaceholderText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReviewSection userReview={undefined} reviews={[]} carID="1" />
      </MockedProvider>,
    );

    const reviewButton = getByText('Review this car');
    fireEvent.click(reviewButton);

    const reviewTextArea = getByPlaceholderText('Add a review to your rating');
    expect(reviewTextArea).toBeTruthy();
  });

  test('adds a review when the "Submit review" button is clicked', async () => {
    const { getByText, getByPlaceholderText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReviewSection userReview={undefined} reviews={[]} carID="1" />
      </MockedProvider>,
    );

    const reviewButton = getByText('Review this car');
    fireEvent.click(reviewButton);

    const reviewTextArea = getByPlaceholderText('Add a review to your rating');
    userEvent.type(reviewTextArea, 'Test review');

    const submitButton = getByText('Submit review');
    userEvent.click(submitButton);

    await waitFor(() => {});
  });
});
