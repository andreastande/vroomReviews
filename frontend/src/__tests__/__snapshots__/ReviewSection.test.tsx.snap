// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewSection Component > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="reviews"
  >
    <h2>
      There are no reviews yet for this car
    </h2>
    <section
      class="current-user-review"
    >
      <p>
         Your review: 
      </p>
      <div>
        <section
          class="starRating"
          data-testid="star-rating"
        >
          <button
            aria-label="1 star rating"
            class="star"
            disabled=""
            style="height: auto; width: 30px; color: rgb(216, 161, 39);"
          >
            <svg
              aria-label="star"
              fill="currentColor"
              role="img"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
              />
            </svg>
          </button>
          <button
            aria-label="2 star rating"
            class="star"
            disabled=""
            style="height: auto; width: 30px; color: rgb(216, 161, 39);"
          >
            <svg
              aria-label="star"
              fill="currentColor"
              role="img"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
              />
            </svg>
          </button>
          <button
            aria-label="3 star rating"
            class="star"
            disabled=""
            style="height: auto; width: 30px; color: rgb(216, 161, 39);"
          >
            <svg
              aria-label="star"
              fill="currentColor"
              role="img"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
              />
            </svg>
          </button>
          <button
            aria-label="4 star rating"
            class="star"
            disabled=""
            style="height: auto; width: 30px; color: rgb(216, 161, 39);"
          >
            <svg
              aria-label="star"
              fill="currentColor"
              role="img"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
              />
            </svg>
          </button>
          <button
            aria-label="5 star rating"
            class="star"
            disabled=""
            style="height: auto; width: 30px; color: rgb(216, 161, 39);"
          >
            <svg
              aria-label="star"
              fill="currentColor"
              role="img"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
              />
            </svg>
          </button>
        </section>
        <p>
          Test review
        </p>
      </div>
      <u
        class="delete-review"
      >
        delete
      </u>
    </section>
  </section>
</DocumentFragment>
`;
